Test/search flow local data source

Noor-Serry opened the pull request:
# Pull Request Template

## Description
This PR includes several improvements and bug fixes related to recent search functionality:

✅ Refactored all data source tests

✅ Refactored the RecentSearchDaoTest. 
hint(*other dao tests fail)

🐛 Fixed a bug where recent searches were displayed from oldest to newest instead of newest to oldest by updating the query to sort by dateAdded in descending order.
---

## Changes Made
List the changes introduced in this pull request:

- Recent search dao && LocalSearchDto
- all unit test for local data source 
- android test for recent search dao

---

## Screenshots (if applicable)
Include screenshots or GIFs to showcase the changes, especially if they impact the UI.

<img width="870" height="210" alt="image" src="https://github.com/user-attachments/assets/ee0b3418-0716-4a7b-a613-c5fdb91e0d96" />

---

## Checklist
Please ensure the following tasks are completed:

- [x] My code follows the code style of this project
- [x] Changes have been tested manually and verified.
- [x] PR includes at most one single feature.

---

## Additional Comments
Add any additional information or context about the pull request here.

Noor-Serry committed:
fix: Add timestamp to recent searches and order by date

This commit introduces a `dateAdded` field to the `LocalSearchDto` to track when a search was made.

Recent searches are now fetched in descending order of their `dateAdded` timestamp, ensuring the most recent searches appear first.

The `deleteRecentSearchByKeywordAndType` function in `RecentSearchLocalSource` has been renamed to `deleteRecentSearchByKeyword` as the type parameter is no longer needed. The corresponding DAO method `deleteSearchByKeyword` has been updated to reflect this change.

Noor-Serry committed:
refactor: Adapt local data sources to new schema and remove unused code

This commit refactors the local data sources and their tests to align with recent schema changes.

Key changes include:
- Removing `SearchMapping` and related test utilities as they are no longer used.
- Updating `createMovie` utility to use `releaseDate` instead of `productionYear` and remove `hasVideo`.
- Modifying `CategoryLocalDataSourceImplTest` to include `storedLanguage` parameter in DAO calls.
- Removing search-related functionality (keyword, searchType) from `MovieLocalDataSourceImpl` and its tests, as this logic is now handled elsewhere.
- Adding tests for new `MovieLocalDataSourceImpl` methods: `insertMovie`, `getPopularMovies`, `getUpcomingMovies`, `getTopRatedMovies`, and methods for adding and deleting these movie lists.
- Simplifying `RecentSearchLocalDataSourceImpl` by removing `searchType`, `storedLanguage`, and `expireDate`. Tests have been updated accordingly.
- Updating `TvShowLocalDataSourceImplTest` to reflect changes in how TV shows and categories are handled.
- Adding new tests for `TvShowLocalDataSourceImpl` methods: `insertTvShow`, `getPopularTvShows`, `getTopRatedTvShows`, and methods for adding and deleting these TV show lists.
- Introducing `WatchHistoryLocalDataSourceImplTest` with tests for adding and retrieving movie and TV show watch history.
- Updating `RecentSearchRepositoryImpl` to remove `type` and `language` parameters from `deleteRecentSearchByKeywordAndType`.
- Adjusting `createLocalTvShowCategoryDto` and `createLocalSearchDto` test utilities to match schema updates.

Noor-Serry committed:
test: update CategoryDaoTest and RecentSearchDaoTest

This commit updates the tests for `CategoryDao` and `RecentSearchDao`.

- **CategoryDaoTest:**
    - Modified tests to pass the `storedLanguage` parameter when calling `getAllMovieCategories` and `getAllTvShowCategories`.
- **RecentSearchDaoTest:**
    - Simplified `upsertRecentSearch` tests.
    - Removed tests related to `SearchType` and expiration, as these are no longer relevant.
    - Updated `deleteSearchByKeyword` test to reflect current functionality.
    - Refactored `ComparableSearchDto` and `toComparable()` to only include `searchKeyword` and `dateAdded`.
- Added `kotlinx-coroutines-test` dependency to `localDatasource/build.gradle.kts`.

saraashraf11 commented on localDatasource/src/androidTest/kotlin/com/amsterdam/localdatasource/daos/CategoryDaoTest.kt:
```diff
@@ -62,14 +64,14 @@ class CategoryDaoTest {
     }
 
     @Test
     fun getAllMovieCategories_shouldReturnEmptyList_whenNoDataInserted() = runTest {
```
In order to increase the readability you could have written the message as `should return ... when ....`, What do you think?

Noor-Serry replied:
no that is not valid in this case because I have more than one function inside the class under the test so that I need to put the name of the function (unit) that I test as prefix

saraashraf11 replied:
I just want you to use ` and write the message inside it which increases the readability

Noor-Serry replied:
I got you now.
but that is applicable  in unit test not  in android test environment.

saraashraf11 commented:
I have only one modification regarding the way of writing the message in the test, as I explained in the comment and as you use in some files.

Noor-Serry commented:
@saraashraf11 
as I explain I need to put the name of the function I test as prefix because I have more than one function inside the class under the test so that I need to distinguish between theme

nabilBouzineDev commented on localDatasource/src/androidTest/kotlin/com/amsterdam/localdatasource/daos/RecentSearchDaoTest.kt:
```diff
@@ -35,120 +33,71 @@ class RecentSearchDaoTest {
-        val search = createLocalSearchDto("barbie", SearchType.BY_KEYWORD)
+        val search = createLocalSearchDto("barbie")
 
         // Act
```
Please change this to When to match the other test cases

Noor-Serry replied:
ok

Noor-Serry committed:
test: update Given-When-Then comments in RecentSearchDaoTest

This commit updates the comments in `RecentSearchDaoTest.kt` to consistently use "When" instead of "Act" in the Given-When-Then pattern for test structure.
